libfwupd/fwupd-device.c

```diff
@@ -254,6 +255,58 @@ fwupd_device_get_guid_default (FwupdDevice *device)
 }
 
 /**
+ * fwupd_device_get_icons:
+ * @device: A #FwupdDevice
+ *
+ * Gets the icon names to use for the device.
+ *
+ * NOTE: Icons specified without a full path are stock icons and should
+ * be loaded from the users icon theme.
+ *
+ * Returns: (element-type utf8) (transfer none): an array of icon names
+ *
+ * Since: 0.9.8
+ **/
+GPtrArray *
+fwupd_device_get_icons (FwupdDevice *device)
+{
+	FwupdDevicePrivate *priv = GET_PRIVATE (device);
+	g_return_val_if_fail (FWUPD_IS_DEVICE (device), NULL);
+	return priv->icons;
+}
+
+static gboolean
+fwupd_device_has_icon (FwupdDevice *device, const gchar *icon)
+{
+	FwupdDevicePrivate *priv = GET_PRIVATE (device);
+	for (guint i = 0; i < priv->icons->len; i++) {
+		const gchar *icon_tmp = g_ptr_array_index (priv->icons, i);
+		if (g_strcmp0 (icon, icon_tmp) == 0)
+			return TRUE;
+	}
+	return FALSE;
+}
+
+/**
+ * fwupd_device_add_icon:
+ * @device: A #FwupdDevice
+ * @icon: the name, e.g. "input-mouse" or "/usr/share/icons/foo.png"
+ *
+ * Adds the icon name if it does not already exist.
+ *
+ * Since: 0.9.8
+ **/
+void
+fwupd_device_add_icon (FwupdDevice *device, const gchar *icon)
+{
+	FwupdDevicePrivate *priv = GET_PRIVATE (device);
+	g_return_if_fail (FWUPD_IS_DEVICE (device));
+	if (fwupd_device_has_icon (device, icon))
```

Comment:
Weren't you going to remove this (and move the code in)?

Reply:
It's unexported and static now.